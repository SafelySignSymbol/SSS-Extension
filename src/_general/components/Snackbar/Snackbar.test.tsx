import { render, screen } from '@testing-library/react'
import { Snackbar, SnackbarType } from '.'

export {}

const test_snackbar_text = 'スナックバーに表示されるテキスト'
const test_snackbar_text_long =
  'テスト用のプログラムでスナックバーに表示されるとても長いテキストです'
describe('/components/Snackbar', () => {
  const Target: React.FC<{
    open: boolean
    text: string
    type: SnackbarType
  }> = ({ open, text, type }) => (
    <Snackbar isOpen={open} snackbarMessage={text} snackbarStatus={type} />
  )

  describe('活性時', () => {
    it('適切にレンダーできる: <DEFAULT>', () => {
      const { container } = render(
        <Target
          open={true}
          text={test_snackbar_text}
          type={SnackbarType.DEFAULT}
        />
      )
      expect(container).toMatchSnapshot()
    })
    it('適切にレンダーできる: <INFO>', () => {
      const { container } = render(
        <Target
          open={true}
          text={test_snackbar_text}
          type={SnackbarType.INFO}
        />
      )
      expect(container).toMatchSnapshot()
    })
    it('適切にレンダーできる: <ERROR>', () => {
      const { container } = render(
        <Target
          open={true}
          text={test_snackbar_text}
          type={SnackbarType.ERROR}
        />
      )
      expect(container).toMatchSnapshot()
    })
    it('適切にレンダーできる: <WARN>', () => {
      const { container } = render(
        <Target
          open={true}
          text={test_snackbar_text}
          type={SnackbarType.WARN}
        />
      )
      expect(container).toMatchSnapshot()
    })
    it('適切にレンダーできる: <SUCCESS>', () => {
      const { container } = render(
        <Target
          open={true}
          text={test_snackbar_text}
          type={SnackbarType.SUCCESS}
        />
      )
      expect(container).toMatchSnapshot()
    })
    it('適切にレンダーできる: 長いテキスト', () => {
      const { container } = render(
        <Target
          open={true}
          text={test_snackbar_text_long}
          type={SnackbarType.DEFAULT}
        />
      )
      expect(container).toMatchSnapshot()
    })
    it('スナックバーに表示されるテキスト', () => {
      render(
        <Target
          open={true}
          text={test_snackbar_text}
          type={SnackbarType.DEFAULT}
        />
      )
      expect(screen.queryByText(test_snackbar_text)).toBeTruthy()
    })
  })
  // describe('非活性時にnullが帰る', () => {
  //   const { container } = render(
  //     <Target open={false} text="" type={SnackbarType.DEFAULT} />
  //   )
  //   console.log(container)
  //   expect(container).toEqual(null)
  // })
})
