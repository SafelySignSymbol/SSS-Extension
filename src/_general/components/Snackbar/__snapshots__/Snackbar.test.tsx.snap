// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/components/Snackbar 活性時 適切にレンダーできる: <DEFAULT> 1`] = `
<div>
  <span
    class="css-33mmzo"
  >
    <div
      class="css-y9ug4i"
      color="#333333"
    >
      <svg
        fill="currentColor"
        height="20px"
        stroke="currentColor"
        stroke-width="0"
        style="color: rgb(238, 238, 238);"
        version="1.1"
        viewBox="0 0 16 16"
        width="20px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M7 4.75c0-0.412 0.338-0.75 0.75-0.75h0.5c0.412 0 0.75 0.338 0.75 0.75v0.5c0 0.412-0.338 0.75-0.75 0.75h-0.5c-0.412 0-0.75-0.338-0.75-0.75v-0.5z"
        />
        <path
          d="M10 12h-4v-1h1v-3h-1v-1h3v4h1z"
        />
        <path
          d="M8 0c-4.418 0-8 3.582-8 8s3.582 8 8 8 8-3.582 8-8-3.582-8-8-8zM8 14.5c-3.59 0-6.5-2.91-6.5-6.5s2.91-6.5 6.5-6.5 6.5 2.91 6.5 6.5-2.91 6.5-6.5 6.5z"
        />
      </svg>
    </div>
    <div
      class="css-15iwusg"
    >
      スナックバーに表示されるテキスト
    </div>
  </span>
</div>
`;

exports[`/components/Snackbar 活性時 適切にレンダーできる: <ERROR> 1`] = `
<div>
  <span
    class="css-33mmzo"
  >
    <div
      class="css-m6igfi"
      color="#CB1742"
    >
      <svg
        fill="currentColor"
        height="20px"
        stroke="currentColor"
        stroke-width="0"
        style="color: rgb(238, 238, 238);"
        version="1.1"
        viewBox="0 0 16 16"
        width="20px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8 1.5c-1.736 0-3.369 0.676-4.596 1.904s-1.904 2.86-1.904 4.596c0 1.736 0.676 3.369 1.904 4.596s2.86 1.904 4.596 1.904c1.736 0 3.369-0.676 4.596-1.904s1.904-2.86 1.904-4.596c0-1.736-0.676-3.369-1.904-4.596s-2.86-1.904-4.596-1.904zM8 0v0c4.418 0 8 3.582 8 8s-3.582 8-8 8c-4.418 0-8-3.582-8-8s3.582-8 8-8zM7 11h2v2h-2zM7 3h2v6h-2z"
        />
      </svg>
    </div>
    <div
      class="css-15iwusg"
    >
      スナックバーに表示されるテキスト
    </div>
  </span>
</div>
`;

exports[`/components/Snackbar 活性時 適切にレンダーできる: <INFO> 1`] = `
<div>
  <span
    class="css-33mmzo"
  >
    <div
      class="css-j4cgke"
      color="#1753CB"
    >
      <svg
        fill="currentColor"
        height="20px"
        stroke="currentColor"
        stroke-width="0"
        style="color: rgb(238, 238, 238);"
        version="1.1"
        viewBox="0 0 16 16"
        width="20px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M7 4.75c0-0.412 0.338-0.75 0.75-0.75h0.5c0.412 0 0.75 0.338 0.75 0.75v0.5c0 0.412-0.338 0.75-0.75 0.75h-0.5c-0.412 0-0.75-0.338-0.75-0.75v-0.5z"
        />
        <path
          d="M10 12h-4v-1h1v-3h-1v-1h3v4h1z"
        />
        <path
          d="M8 0c-4.418 0-8 3.582-8 8s3.582 8 8 8 8-3.582 8-8-3.582-8-8-8zM8 14.5c-3.59 0-6.5-2.91-6.5-6.5s2.91-6.5 6.5-6.5 6.5 2.91 6.5 6.5-2.91 6.5-6.5 6.5z"
        />
      </svg>
    </div>
    <div
      class="css-15iwusg"
    >
      スナックバーに表示されるテキスト
    </div>
  </span>
</div>
`;

exports[`/components/Snackbar 活性時 適切にレンダーできる: <SUCCESS> 1`] = `
<div>
  <span
    class="css-33mmzo"
  >
    <div
      class="css-xye49e"
      color="#46CB17"
    >
      <svg
        fill="currentColor"
        height="20px"
        stroke="currentColor"
        stroke-width="0"
        style="color: rgb(238, 238, 238);"
        version="1.1"
        viewBox="0 0 16 16"
        width="20px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M13.5 2l-7.5 7.5-3.5-3.5-2.5 2.5 6 6 10-10z"
        />
      </svg>
    </div>
    <div
      class="css-15iwusg"
    >
      スナックバーに表示されるテキスト
    </div>
  </span>
</div>
`;

exports[`/components/Snackbar 活性時 適切にレンダーできる: <WARN> 1`] = `
<div>
  <span
    class="css-33mmzo"
  >
    <div
      class="css-ykjjq7"
      color="#CAAC16"
    >
      <svg
        fill="currentColor"
        height="20px"
        stroke="currentColor"
        stroke-width="0"
        style="color: rgb(238, 238, 238);"
        version="1.1"
        viewBox="0 0 16 16"
        width="20px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8 1.45l6.705 13.363h-13.409l6.705-13.363zM8 0c-0.345 0-0.69 0.233-0.951 0.698l-6.829 13.611c-0.523 0.93-0.078 1.691 0.989 1.691h13.583c1.067 0 1.512-0.761 0.989-1.691h0l-6.829-13.611c-0.262-0.465-0.606-0.698-0.951-0.698v0z"
        />
        <path
          d="M9 13c0 0.552-0.448 1-1 1s-1-0.448-1-1c0-0.552 0.448-1 1-1s1 0.448 1 1z"
        />
        <path
          d="M8 11c-0.552 0-1-0.448-1-1v-3c0-0.552 0.448-1 1-1s1 0.448 1 1v3c0 0.552-0.448 1-1 1z"
        />
      </svg>
    </div>
    <div
      class="css-15iwusg"
    >
      スナックバーに表示されるテキスト
    </div>
  </span>
</div>
`;

exports[`/components/Snackbar 活性時 適切にレンダーできる: 長いテキスト 1`] = `
<div>
  <span
    class="css-33mmzo"
  >
    <div
      class="css-y9ug4i"
      color="#333333"
    >
      <svg
        fill="currentColor"
        height="20px"
        stroke="currentColor"
        stroke-width="0"
        style="color: rgb(238, 238, 238);"
        version="1.1"
        viewBox="0 0 16 16"
        width="20px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M7 4.75c0-0.412 0.338-0.75 0.75-0.75h0.5c0.412 0 0.75 0.338 0.75 0.75v0.5c0 0.412-0.338 0.75-0.75 0.75h-0.5c-0.412 0-0.75-0.338-0.75-0.75v-0.5z"
        />
        <path
          d="M10 12h-4v-1h1v-3h-1v-1h3v4h1z"
        />
        <path
          d="M8 0c-4.418 0-8 3.582-8 8s3.582 8 8 8 8-3.582 8-8-3.582-8-8-8zM8 14.5c-3.59 0-6.5-2.91-6.5-6.5s2.91-6.5 6.5-6.5 6.5 2.91 6.5 6.5-2.91 6.5-6.5 6.5z"
        />
      </svg>
    </div>
    <div
      class="css-15iwusg"
    >
      テスト用のプログラムでスナックバーに表示されるとても長いテキストです
    </div>
  </span>
</div>
`;
